R/test-server.R

```diff
@@ -50,20 +50,21 @@ isModuleServer <- function(x) {
 #'   # Any additional arguments, below, are passed along to the module.
 #' })
 #' @export
-testServer <- function(app = NULL, expr, args = list()) {
+testServer <- function(app = NULL, expr, args = list(), session = MockShinySession$new()) {
 
   require(shiny)
 
   quosure <- rlang::enquo(expr)
-  session <- getDefaultReactiveDomain()
 
-  if (inherits(session, "MockShinySession"))
-    stop("Test expressions may not call testServer()")
-  if (inherits(session, "session_proxy")
-      && inherits(get("parent", envir = session), "MockShinySession"))
-    stop("Modules may not call testServer()")
+  local({
```

Comment:
Can remove `local` because it will make the stack trace more confusing.

Also, a comment explaining why this is here would be helpful.